### PR TITLE
Add token and badge styles from V2

### PR DESCRIPTION
**Pull request by daverau-optimizely**
Added these from v2 without much modification. @kwalker3690 had a good point — that we should rename these classes down the road and do a search/replace update in the app for `.badge__primary` to `.badge--primary` — something to do in the near future.

Here's what the docs will look like:

screenshot 2016-06-17 14 24 26

screenshot 2016-06-17 14 24 52

### CODE REVIEW

File: src/oui/partials/components/_badge.scss [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    height: map-fetch($badge, size);
+    line-height: map-fetch($badge, size);
+    background: #B4B7B9;
+    color: map-fetch($color, ui, white);
+    min-width: 16px;
+    text-align: center;
+    margin-right: 5px;
+    border-radius: map-fetch($badge, size);
+    font-size: 10px;
+    padding: 0 5px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: spacer(0.5)

File: src/oui/partials/components/_badge.scss [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  > li {
+    height: map-fetch($badge, size);
+    line-height: map-fetch($badge, size);
+    background: #B4B7B9;
+    color: map-fetch($color, ui, white);
+    min-width: 16px;
+    text-align: center;
+    margin-right: 5px;
+    border-radius: map-fetch($badge, size);
+    font-size: 10px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: Can this be a variable in the token map?

File: src/oui/partials/components/_badge.scss [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  white-space: nowrap;
+
+  > li {
+    height: map-fetch($badge, size);
+    line-height: map-fetch($badge, size);
+    background: #B4B7B9;
+    color: map-fetch($color, ui, white);
+    min-width: 16px;
+    text-align: center;
+    margin-right: 5px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: spacer(0.5)

File: src/oui/partials/components/_badge.scss [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+  display: flex;
+  align-items: center;
+  white-space: nowrap;
+
+  > li {
+    height: map-fetch($badge, size);
+    line-height: map-fetch($badge, size);
+    background: #B4B7B9;
+    color: map-fetch($color, ui, white);
+    min-width: 16px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: Can this be a variable in the token map

File: src/oui/partials/components/_badge.scss [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+.#{$namespace}badge {
+  display: flex;
+  align-items: center;
+  white-space: nowrap;
+
+  > li {
+    height: map-fetch($badge, size);
+    line-height: map-fetch($badge, size);
+    background: #B4B7B9;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: Variable?

File: src/oui/partials/components/_token.scss [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+.#{$namespace}token {
+  display: inline-flex;
+  align-items: center;
+
+  color: map-fetch($color, text, white);
+  line-height: 1;
+  padding: spacer(0.5) spacer(1);
+  border-radius: 2px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: map-fetch($border-radius, base)

File: src/oui/partials/components/_badge.scss (hunk 4 above)
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: Oh or in a badge map actually

File: src/oui/partials/components/_badge.scss (hunk 2 above)
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: *badge

File: src/oui/partials/components/_badge.scss [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+    color: map-fetch($color, ui, white);
+    min-width: 16px;
+    text-align: center;
+    margin-right: 5px;
+    border-radius: map-fetch($badge, size);
+    font-size: 10px;
+    padding: 0 5px;
+    font-weight: 700;
+    letter-spacing: 0.5px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_badge.scss]: Variable in badge map would be better

File: src/oui/partials/components/_token.scss [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  flex: none;
+  min-width: 0;
+  min-height: 0;
+  padding-left: map-fetch($token, padding) * 2;
+}
+
+.#{$namespace}token__number {
+  color: map-fetch($color, text, light);
+  font-size: map-fetch($font, size, milli);
+  min-width: 10px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Add to variable map

File: src/oui/partials/components/_token.scss [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+.#{$namespace}token__description {
+  font-size: map-fetch($font, size, micro);
+  margin-top: spacer(0.5);
+}
+
+.#{$namespace}token__close {
+  color: map-fetch($color, ui, white);
+  height: 12px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Put these in the variable map

File: src/oui/partials/components/_token.scss [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+  color: map-fetch($color, text, light);
+  font-size: map-fetch($font, size, milli);
+  min-width: 10px;
+}
+
+.#{$namespace}token__move {
+  transform: rotate(90deg);
+
+  color: darken(map-fetch($color, ui, base), 10);
+  margin-left: -2px;
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Why is this needed? Maybe add a note if it is necessary to keep

File: src/oui/partials/components/_token.scss [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+
+.#{$namespace}token__number {
+  color: map-fetch($color, text, light);
+  font-size: map-fetch($font, size, milli);
+  min-width: map-fetch($token, token__number, min-width);
+}
+
+.#{$namespace}token__move {
+  transform: rotate(90deg);
+
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Nit: don't need a space here

File: src/oui/partials/components/_token.scss [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+  border: 1px solid map-fetch($color, ui, base);
+  background: map-fetch($color, background, light);
+  padding: map-fetch($token, padding);
+  border-radius: map-fetch($token, border-radius);
+}
+
+.#{$namespace}token {
+  display: inline-flex;
+  align-items: center;
+
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Nit: remove line

File: src/oui/partials/components/_token.scss [hunk 13]
```diff
@@ -0,0 +1,92 @@
...
+///           <use xmlns:xlink="http://www.w3.org/1999/xlink" xlink:href="#close-16"></use>
+///         </svg>
+///       </a>
+///     </div>
+///   </div>
+////
+
+.#{$namespace}token-wrap {
+  display: inline-flex;
+
```
Comment (kwalker3690) [inline src/oui/partials/components/_token.scss]: Nit: remove line

**Comment by kwalker3690** [comment]
A few nits but other than that LGTM! Thanks for doing this!

**Comment by daverau-optimizely** [comment]
fixes #417 and #418
